pegged/peg.d

```diff
@@ -1760,11 +1760,13 @@ template or(rules...) if (rules.length > 0)
 }
 
 auto getUpto(scope ParseTree[] children, size_t minFailedLength) pure nothrow {
-    auto ret = children.filter!(r => max(r.end, r.failEnd) >= minFailedLength).array();
-
+    import std.range : iota;
     // FIXME: .array() garantees a copy of the elements. Compiler scope
     // inference issue: https://issues.dlang.org/show_bug.cgi?id=20674
-    return ret.dup;
+    return children.length.iota
```

Comment:
this doesn't seem to work. It's yielding a very weird error tho: `.array` is `@system` with some of these lambdas. Anyway, I would keep `.dup` as it does an efficient memcpy rather than iterating field by field.